GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/SpaceViewModel.cs

```diff
@@ -56,36 +82,124 @@ private void OnGalaxyInteraction(ClassificationRingNotifier RingNotifier)
+            PrepareForNewPosition();
+
+            if (CurrentGalaxies.Count == 0)
+            {
+                double NewBounds = SpaceNavigation.RA + (RaRange / 2);
+                SpacePoint newCenter = _localDBService.FindNextAscendingRa(NewBounds);
+                SpaceNavigation.RA = newCenter.RightAscension;
+                SpaceNavigation.DEC = newCenter.Declination;
+                PrepareForNewPosition();
+            }
+        }
+
+        private void OnMoveViewSouth(object obj)
+        {
+            SpaceNavigation.DEC -= DecRange;
+            PrepareForNewPosition();
+
+            if (CurrentGalaxies.Count == 0)
+            {
+                double NewBounds = SpaceNavigation.DEC - (DecRange / 2);
+                SpacePoint newCenter = _localDBService.FindNextDescendingDec(NewBounds);
+                SpaceNavigation.RA = newCenter.RightAscension;
+                SpaceNavigation.DEC = newCenter.Declination;
+                PrepareForNewPosition();
+            }
         }
 
-        private async Task GetSubjectsAsync()
+        private void OnMoveViewEast(object obj)
         {
-            NameValueCollection query = new NameValueCollection();
-            query.Add("workflow_id", Config.WorkflowId);
-            query.Add("page_size", "25");
+            SpaceNavigation.RA -= RaRange;
+            PrepareForNewPosition();
 
-            var GalaxyList = await _panoptesService.GetSubjectsAsync("queued", query);
-            foreach (Subject subject in GalaxyList)
+            if (CurrentGalaxies.Count == 0)
             {
-                TableSubject Galaxy = new TableSubject(subject, RA, DEC);
-                CurrentGalaxies.Add(Galaxy);
+                double NewBounds = SpaceNavigation.RA - (RaRange / 2);
+                SpacePoint newCenter = _localDBService.FindNextDescendingRa(NewBounds);
+                SpaceNavigation.RA = newCenter.RightAscension;
+                SpaceNavigation.DEC = newCenter.Declination;
+                PrepareForNewPosition();
             }
         }
+
+        private void OnMoveViewNorth(object obj)
+        {
+            SpaceNavigation.DEC += DecRange;
+            PrepareForNewPosition();
+
+            if (CurrentGalaxies.Count == 0)
+            {
+                double NewBounds = SpaceNavigation.DEC + (DecRange / 2);
+                SpacePoint newCenter = _localDBService.FindNextAscendingDec(NewBounds);
+                SpaceNavigation.RA = newCenter.RightAscension;
+                SpaceNavigation.DEC = newCenter.Declination;
+                PrepareForNewPosition();
+            }
+        }
+
+        private void GetSpaceCutout()
+        {
+            double WidenedPlateScale = 1.8;
+            SpaceCutoutUrl = $"http://skyserver.sdss.org/dr14/SkyServerWS/ImgCutout/getjpeg?ra={SpaceNavigation.RA}&dec={SpaceNavigation.DEC}&width=1248&height=432&scale={WidenedPlateScale}";
+        }
+
+        private void PrepareForNewPosition()
```

Comment:
For some reason this function name seems odd to me. I'm not sure I have a suggestion for a better name, but that's because I think I don't have a good understanding of what the function does.

Reply:
In general, I think it would improve clarity to make these functions less stateful. That is, instead of them mutating state, having the function take a `SpaceNavigation` and return a `Galaxy`. That way it's easier to track when state is actually changing.

Reply:
Or at least change the name of the function to something like
 `UpdateCurrentGalaxyWithNavigation (SpaceNavigation Navigation)`

Reply:
This might require some adjustment. `SpaceNavigation` is currently a global constant, but that might be a code smell. I'll make this change as the last commit on the changes I add, and we can decide from there if the change helps strengthen the code.